logstash-core/lib/logstash/settings.rb

```diff
@@ -832,8 +864,7 @@ def initialize(canonical_proxy, alias_name)
         @canonical_proxy = canonical_proxy
 
         clone = @canonical_proxy.canonical_setting.clone
-        clone.instance_variable_set(:@name, alias_name)
-        clone.instance_variable_set(:@default, nil)
```

Comment:
Note for reviewer:

nulling the default value is already done by the `deprecate` method when create the new instance of the Java `Setting` class.